dynatrace_health/stackstate_checks/dynatrace_health/dynatrace_health.py

```
@@ -186,15 +188,18 @@ def _collect_events(self, dynatrace_client, instance_info):
         event_limit_reached = None
         try:
             while events_response:
-                events = events_response.get('events', [])
+                if type(events_response) is dict:
+                    events = events_response.get('events', [])
+                else:
+                    events = events_response
```

this is pretty much fail-silently, if the event_respons eis not what we expect. i'd rather throw/crash out here to not silently discard unexpected responses.

---

in what scenario was this case necessary?

---

I can't remember, this was an issue with the topology api, it's more pre-emptive than an actual tested case.  The thing is, it did happen once or twice, and adding this if-statement resolved it.  I don't see it as failing silently, since the code executes normally after this.  Think of it as hardening.